ion-schema/src/isl/isl_type_reference.rs

```diff
@@ -11,11 +11,64 @@ use ion_rs::value::reader::{element_reader, ElementReader};
 use ion_rs::value::{IonElement, IonStruct};
 use ion_rs::IonType;
 
+/// Provides public facing APIs for constructing ISL type references programmatically for ISL 1.0
+pub mod v_1_0 {
```

Comment:
I don't think it is possible to provide something like `use ion-schema-rust::isl::v_1_0::*;` because this means that all the structs `IslType`, `IslConstraint` are defined within separate module `v_1_0` or `v_2_0`. This leads to 2 different implementations of `IslType` and `IslConstraint`.  I was trying not to create duplicate structures which are doing the same things.

```rust
mod v_1_0 {
  struct IslType {
     ...
  }
}

mod v_2_0 {
  struct IslType {
     ...
  }
}
```